### PR TITLE
Remove extra code previously required in the file for_loop_impl.h

### PR DESCRIPTION
**Pull request by SergeyKopienko**

### CODE REVIEW

**Review by MikeDvorskiy — APPROVED** [review]
LGTM.
But we have to be sure that all tests are OK, for_loop test in particular.

File: include/oneapi/dpl/pstl/experimental/internal/for_loop.h [hunk 1]
```diff
@@ -65,6 +59,9 @@ template <typename _ExecutionPolicy, typename _Ip, typename _Sp, typename... _Re
 void
 for_loop_strided(_ExecutionPolicy&& __exec, type_identity_t<_Ip> __start, _Ip __finish, _Sp __stride, _Rest&&... __rest)
 {
+    static_assert(oneapi::dpl::__internal::__is_host_execution_policy<::std::decay_t<_ExecutionPolicy>>::value,
+                  "for_loop_strided has been implemented for host policies only");
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/pstl/experimental/internal/for_loop.h:63]: => "for_loop_strided is implemented for host policies only"

File: include/oneapi/dpl/pstl/experimental/internal/for_loop_impl.h [hunk 2]
```diff
@@ -161,27 +161,27 @@ class __reduction_pack
...
 
-// Sequenced version of for_loop_n
+// Sequenced + vectorized version of for_loop_n
 template <class _Tag, typename _ExecutionPolicy, typename _Ip, typename _Size, typename _Function, typename... _Rest>
 void
 __pattern_for_loop_n(_Tag, _ExecutionPolicy&&, _Ip __first, _Size __n, _Function __f, __single_stride_type,
                      _Rest&&... __rest) noexcept
 {
-    static_assert(__is_backend_tag_serial_v<_Tag> || __is_backend_tag_parallel_forward_v<_Tag>);
+    static_assert(__is_backend_tag_v<_Tag>);
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/pstl/experimental/internal/for_loop_impl.h:170]: It seems that should be `static_assert(__is_backend_tag_serial_v<_Tag> || __is_backend_tag_parallel_forward_v<_Tag>);`

File: test/parallel_api/experimental/for_loop.pass.cpp [hunk 3]
```diff
@@ -296,8 +296,8 @@ test_for_loop()
         Sequence<T> in_out(n, Gen<T>());
         Sequence<T> expected = in_out;
 
-        invoke_on_all_policies<>()(test_for_loop_impl(), in_out.begin(), in_out.end(), expected.begin(), expected.end(),
-                                   in_out.size());
+        invoke_on_all_host_policies()(test_for_loop_impl(), in_out.begin(), in_out.end(), expected.begin(),
```
Comment (MikeDvorskiy) [inline test/parallel_api/experimental/for_loop.pass.cpp:299]: Lets add a comment above that we have just the host implementation of `for_loop` and do the test just for the host policies.

File: include/oneapi/dpl/pstl/experimental/internal/for_loop_impl.h (hunk 2 above)
Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/experimental/internal/for_loop_impl.h]: Fixed

File: test/parallel_api/experimental/for_loop.pass.cpp (hunk 3 above)
Comment (SergeyKopienko) [reply test/parallel_api/experimental/for_loop.pass.cpp:300]: Fixed

**Pull request closed without merging**